src/rules/order.js

```diff
@@ -615,6 +615,10 @@ module.exports = {
       TSImportEqualsDeclaration: function handleImports(node) {
         let name
         let type
+        // skip "export import"s
+        if (node.isExport) {
+          return
+        }
         if (node.moduleReference.type === 'TSExternalModuleReference') {
           name = node.moduleReference.expression.value
           type = 'import'
```

Comment:
Sadly I can't create a suggestion because the `registerNode` is too far appart from your actual change... however:
I'd recommend you to change this piece of code to...

```ts
    if (!node.isExport && moduleReference.type === 'TSExternalModuleReference') {
        registerNode(
            context,
            node,
            node.moduleReference.expression.value,
            'import',
            raunks,
            imported,
            pathGroupdExcludedImportTypes
        )
    }

Reply:
Also please notice, that these pieces of code:
https://github.com/benmosher/eslint-plugin-import/blob/c36429bf6cb80e6ca75696de031a58927bba8db5/src/rules/order.js#L315-L319
https://github.com/benmosher/eslint-plugin-import/blob/c36429bf6cb80e6ca75696de031a58927bba8db5/src/rules/order.js#L327-L329
Is not necessary anymore anymore after applying this change, because `import:object` won't exist anymore in this rule.

Reply:
There may still be `import x = object`, so this needs to be kept.

Reply:
Though, that's true, but forcing object-literals to be alphabetized might cause errors as elaborated sooner.
I honestly don't really know what to do about object-imports 😅

Reply:
@manuth Maybe we should simply set their rank to a same, super-large value...
Anyway this should be in another PR.